test/conformance/gateway_expermimental_conformance_test.go

```diff
@@ -0,0 +1,84 @@
+//go:build conformance_tests && experimental
+
+package conformance
+
+import (
+	"os"
+	"path"
+	"strings"
+	"testing"
+
+	"github.com/blang/semver/v4"
+	"github.com/stretchr/testify/require"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"sigs.k8s.io/gateway-api/conformance/apis/v1alpha1"
+	"sigs.k8s.io/gateway-api/conformance/tests"
+	"sigs.k8s.io/gateway-api/conformance/utils/suite"
+	"sigs.k8s.io/yaml"
+
+	"github.com/kong/kubernetes-ingress-controller/v2/internal/manager/metadata"
+	"github.com/kong/kubernetes-ingress-controller/v2/test/internal/testenv"
+)
+
+func TestGatewayExperimentalConformance(t *testing.T) {
+	// Experimental conformance tests passes only when expression routes are enabled
+	// (KONG_TEST_EXPRESSION_ROUTES='true' set automatically in make test.conformance-experimental)
+	// because it supports more features and it's the desired way to use Kong in the future.
+	client, gatewayClassName := prepareEnvForGatewayConformanceTests(t)
+	// Release is expected to have format v2.8.1 (when run on tagged commit) or v2.8.1-731-g4a32cf0b.
+	_, err := semver.Parse(strings.TrimPrefix(metadata.Release, "v"))
```

Comment:
Experimental uses this version only to put it in YAML report, previous version of conformance tests didn't have such feature

Reply:
I see 😄 Experimental is called experimental, because introduced many new features, and one of them is reporting capability that is not 100% finished yet. We've started [publishing reports](ttps://github.com/kubernetes-sigs/gateway-api/tree/main/conformance/reports) since `0.7.1`, but manually.

This PR introduces experimental because we want to start to use it to generate those reports. Be ready when the experimental becomes stable. Next, introduce automation to make PRs with reports directly from CI. The Gateway API team still works and changes stuff, so we have to pay attention

Reply:
That experimental becomes stable soon. Experimental means that they want to make some big changes not so much in tests, but in the whole machinery, like the ability to generate reports in a predefined format (the old one can't do it)